sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt

```diff
@@ -114,58 +118,72 @@ class AppSyncAccountRepository @Inject constructor(
         }
     }
 
-    override fun processCode(stringCode: String): Result<Boolean> {
-        val decodedCode: String? = kotlin.runCatching {
-            return@runCatching stringCode.decodeB64()
-        }.getOrNull()
-        if (decodedCode == null) {
-            Timber.w("Failed while b64 decoding barcode; barcode is unusable")
-            return Error(code = INVALID_CODE.code, reason = "Failed to decode code")
-        }
-
-        kotlin.runCatching {
-            Adapters.recoveryCodeAdapter.fromJson(decodedCode)?.recovery
-        }.getOrNull()?.let {
-            Timber.d("Sync: code is a recovery code")
-            return login(it)
-        }
+    override fun processCode(code: CodeType): Result<Boolean> {
```

Comment:
now takes in the parsed code so we don’t have to parse it from JSON twice any more